iop/sio/mx4sio_bd/src/mx4sio.c

```diff
@@ -875,15 +912,15 @@ static void sd_detect_thread(void *arg)
     M_PRINTF("card detection thread running\n");
 
     while (1) {
-        // Sleep for 1 second
-        DelayThread(1000 * 1000);
-
         M_DEBUG("Check card, inserted=%d, used=%d\n", card_inserted, card_used);
 
         // Detect card if it has not been used recently
         if (card_used == 0)
             sd_detect();
         card_used = 0;
+
+        // Sleep for 1 second
+        DelayThread(1000 * 1000);
```

Comment:
Originally this delay was right there where you put it, but there where issues with this. By putting it at the beginning of the loop there is an (intentional) delay before the first poll starts after the thread is created.

Is there a reason why this is changed, besides being a little faster, has this also been proven stable?

Reply:
With the delay in its original location my mx4sio card would take a long time to mount in OPL. This particular thread runs at low priority, and from my debugging would take around 5 seconds before it got scheduled in even though the wait is supposed to be 1 second. Since the comment didn't note that the location of the wait was intentional I moved it, which resulted in the mx4sio card getting mounted immediately.

I have not experienced any issues after moving it, but if there previously were issues regarding it's location then a larger sample size would be needed to say if it's stable or not.

Reply:
Your debugging must have been using 100% cpu for 5 seconds with a higher priority if it took so long.

The whole card detection thing is terrible and in need of a rewrite if you're interested? When there is no card inserted, but the mx4sio driver is loaded, this loop will hang the sio2 bus every second, causing OPL (or any other application) to become unresponsive since controller input is handled over the same sio2 bus.

But anyway, this change can couse side effect and it's off-topic for this PR, so I think it's better to remove it.